src/content/blog/astro-480.mdx

```diff
@@ -0,0 +1,177 @@
+publishDate: "May 9, 2024"
+authors:
+  - erika
+  - ema
+  - matthew
+  - nate
+  - bjorn
+  - sarah
+coverImage: "/src/content/blog/_images/astro-480/header_astro_4_8.webp"
+socialImage: "/src/content/blog/_images/astro-480/og_image_4_8.webp"
+lang: "en"
+---
+
+import BlogSummary from "/src/components/BlogSummary.astro";
+
+<BlogSummary>
+  Astro 4.8 is out now! This release includes experimental support for Astro
+  actions and request rewriting, performance improvements, and more.
+</BlogSummary>
+
+Full release highlights include:
+
+- [**Experimental: Astro Actions**](#experimental-astro-actions)
+- [**Experimental: Request Rewriting**](#experimental-request-rewriting)
+- [**Performance improvements**](#performance-improvements)
+- [**Ability to define multiple routes with the same entrypoint**](#ability-to-define-multiple-routes-with-the-same-entrypoint)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Astro Actions
+
+This release includes the first preview for a long awaited feature: Astro actions! Actions make it easy to define and call backend functions with full type-safety from your client code.
+
+To use this feature, [add a server build output](https://docs.astro.build/en/basics/rendering-modes/#on-demand-rendered) and enable the `experimental.actions` option in your Astro config:
+
+```js
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  output: "hybrid", // or 'server'
+  experimental: {
+    actions: true,
+  },
+});
+```
+
+Then, define actions in a `src/actions/index.ts` file using `defineAction()`. You can define actions that accept JSON or form requests, and the `handler` function will be called with your type-safe input.
+
+Plus, no more casting `formData.get()` results. Astro will automatically parse form requests to objects using your Zod schema. Here's an example of a newsletter signup action:
+
+```ts
+// src/actions/index.ts
+import { defineAction, z } from "astro:actions";
+
+export const server = {
+  newsletter: defineAction({
+    accept: "form",
+    input: z.object({
+      email: z.string().email(),
+      receivePromo: z.boolean(),
+    }),
+    handler: async ({ email, receivePromo }) => {
+      // call a mailing service, or store to a database
+      return { success: true };
+    },
+  }),
+};
+```
+
+You can call actions from any client component using the `actions` object from `astro:actions`. You can pass a type-safe object when using JSON, or a `FormData` object when using `accept: 'form'`. You can also use `getActionProps()` for progressive enhancement:
```

Comment:
```suggestion
You can call actions from any on-demand rendered client component using the `actions` object from `astro:actions`. You can pass a type-safe object when using JSON, or a `FormData` object when using `accept: 'form'`. You can also use `getActionProps()` for progressive enhancement:
```
Just to match saying that one of the server modes is required! This is where it matters!

Reply:
Ah, that's actually incorrect! You can _call_ actions from statically rendered pages. You only need server rendering for the actions themselves.